lib/node_modules/@stdlib/lapack/base/dlacpy/src/dlacpy.c

```diff
@@ -0,0 +1,212 @@
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+#include "stdlib/lapack/base/dlacpy.h"
+#include "stdlib/blas/base/shared.h"
+#include "stdlib/strided/base/stride2offset.h"
+#include "stdlib/math/base/special/fast/min.h"
+#include "stdlib/ndarray/base/assert/is_row_major.h"
+
+/**
+* Copies all of a matrix `A` to another matrix `B`.
+*
+* @param M         number of rows in matrix `A`
+* @param N         number of columns in matrix `A`
+* @param A         input matrix
+* @param strideA1  stride of the first dimension of `A`
+* @param strideA2  stride of the second dimension of `A`
+* @param offsetA   starting index for `A`
+* @param B         output matrix
+* @param strideB1  stride of the first dimension of `B`
+* @param strideB2  stride of the second dimension of `B`
+* @param offsetB   starting index for `B`
+*/
+void API_SUFFIX(c_copy_all)( const CBLAS_INT M, const CBLAS_INT N, const double *A, const CBLAS_INT strideA1, const CBLAS_INT strideA2, const CBLAS_INT offsetA, double *B, const CBLAS_INT strideB1, const CBLAS_INT strideB2, const CBLAS_INT offsetB ) {
+	CBLAS_INT ia;
+	CBLAS_INT ib;
+	CBLAS_INT i0;
+	CBLAS_INT i1;
+
+	ia = offsetA;
+	ib = offsetB;
+	for ( i1 = 0; i1 < M; i1++ ) {
+		for ( i0 = 0; i0 < N; i0++ ) {
+			B[ ib+(i0*strideB2) ] = A[ ia+(i0*strideA2) ];
+		}
+		ia += strideA1;
+		ib += strideB1;
+	}
+	return;
+}
+
+/**
+* Copies all of a matrix `A` to another matrix `B`.
+*
+* @param M         number of rows in matrix `A`
+* @param N         number of columns in matrix `A`
+* @param A         input matrix
+* @param strideA1  stride of the first dimension of `A`
+* @param strideA2  stride of the second dimension of `A`
+* @param offsetA   starting index for `A`
+* @param B         output matrix
+* @param strideB1  stride of the first dimension of `B`
+* @param strideB2  stride of the second dimension of `B`
+* @param offsetB   starting index for `B`
+*/
+void API_SUFFIX(c_copy_upper)( const CBLAS_INT M, const CBLAS_INT N, const double *A, const CBLAS_INT strideA1, const CBLAS_INT strideA2, const CBLAS_INT offsetA, double *B, const CBLAS_INT strideB1, const CBLAS_INT strideB2, const CBLAS_INT offsetB ) {
+	CBLAS_INT ia;
+	CBLAS_INT ib;
+	CBLAS_INT i0;
+	CBLAS_INT i1;
+
+	CBLAS_INT strides[] = { strideA1, strideA2 };
+
+	ia = offsetA;
+	ib = offsetB;
+	if ( stdlib_ndarray_is_row_major( 2, strides ) ) {
+		for ( i1 = 0; i1 < M; i1++ ) {
+			for ( i0 = i1; i0 < N; i0++ ) {
+				B[ ib+(i0*strideB2) ] = A[ ia+(i0*strideA2) ];
+			}
+			ia += strideA1;
+			ib += strideB1;
+		}
+		return;
+	}
+
+	for ( i1 = 0; i1 < N; i1++ ) {
+		for ( i0 = 0; i0 <= stdlib_base_fast_min( i1, M-1 ); i0++ ) {
```

Comment:
You shouldn't be using `fast_min`. That is for doubles. Instead, either define an internal function or a macro for returning the min of two integers.